### PR TITLE
mod: update btcwallet's version

### LINKED ISSUE #7675: [bug]: high cpu usage
### Background

My 0.16.2 node started using lots of CPU (100% of one core consistently), which is unusual behavior. Profile available at https://c-otto.de/lnd/profile

Please let me know if you need any configuration details or anything from the logs.

cpu_1d

### Your environment

* version of `lnd`: v0.16.2-beta
* which operating system (`uname -a` on *Nix): Linux server 5.10.0-13-amd64 #<i></i>1 SMP Debian 5.10.106-1 (2022-03-17) x86_64 GNU/Linux
* version of `btcd`, `bitcoind`, or other backend: v24.0.1

### PR DESCRIPTION
**Pull request by yyforyongyu**
Update btcwallet's version to include an optimized mempool lookup.

Fixes #7675

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -187,3 +187,5 @@ replace github.com/gogo/protobuf => github.com/gogo/protobuf v1.3.2
 go 1.19
 
 retract v0.0.2
+
+replace github.com/btcsuite/btcwallet v0.16.9 => github.com/yyforyongyu/btcwallet v0.16.8-0.20230509120640-2374a7fddef8
```
Comment (kirito41dd) [inline go.mod:191]: hello, why not merge to btcsuite/btcwallet first?

Comment (yyforyongyu) [reply go.mod:191]: [this PR](https://github.com/btcsuite/btcwallet/pull/864) is under review. Once merged, we'll then update the mod here to merge this one.

**Comment by Roasbeef** [comment]
Can be rebased now with the dep PR merged: https://github.com/btcsuite/btcwallet/pull/864

**Review by Roasbeef — APPROVED** [review]
LGTM 🏒

**Review by positiveblue — APPROVED** [review]
:+1:

**Review by guggero — APPROVED** [review]

**Pull request merged**